use-cases/NIC-passthrough-and-Kata.md

```diff
@@ -0,0 +1,292 @@
+
+For some use cases, the container demands a direct link to the physical port of the host device, for example in a situation were the container is required to route high BW traffic without having support for acceleration such as SR-IOV.
+
+#### In this guide:
+
+* [Before you start (Restrictions, Requirements and Assumptions)](#before-you-start-restrictions-requirements-and-assumptions)
+* [Pass a physical NIC into a Kata Container](#Pass-a-physical-NIC-into-a-Kata-Container)
+  * [Part 1 – The Host](#part-1--the-host)
+  * [Part 2 – The Container](#part-2--the-container)
+* [Test Your Setup](#Test-Your-Setup)
+
+## Before you start (Restrictions, Requirements and Assumptions)  
+
+The method described in this guide relays on your system [supporting  IOMMU.]( https://en.wikipedia.org/wiki/Input–output_memory_management_unit#Published_specifications) before you start, make sure that your system has this attribute. If it does not, you should look for a different solution then the one presented here. 
+
+If you are uncertain about your system’s support, the first two steps in [Part 1 – The Host](# Part-1-–-The-Host) should help you figure this out. 
+
+ In my setup, I used a server with *Intel(R) Xeon(R) CPU D-1548 @ 2.00GHz* and [support for VT-d](https://en.wikipedia.org/wiki/X86_virtualization#Intel-VT-d), Operated by Ubuntu 18.04.3 LTS. The general steps proposed is this guide should match other hardware and OS with some changes.  
+
+This guide’s starting point is a system with an installed and working *Docker* and *Kata Containers*.
+
+If that is not the case for you:
+
+\-    [How to install Kata Containers]( https://github.com/kata-containers/documentation/tree/master/install)
+
+\-    [How to install Docker]( https://github.com/kata-containers/documentation/blob/master/install/docker/ubuntu-docker-install.md) 
+
+
+
+## Pass a physical NIC into a Kata Container 
+
+### Part 1 – The Host
+
+**Note:** all the next commands are run as a super-user. Use `$ sudo -i` to become a SU.
+
+
+
+#### 1. Configure system to load with IOMMU support.
+
+   Edit grub file ( */etc/default/grub* )
+
+```bash
+vi /etc/default/grub
+```
+
+And add ` intel_iommu=on iommu=pt` To the **GRUB_CMDLINE_LINUX** line 
+
+> GRUB_CMDLINE_LINUX=" … quiet intel_iommu=on iommu=pt"
+
+
+
+If using UEFI: (like I do) 
+
+/TODO – Change to Ubuntu
+
+```bash
+grub2-mkconfig -o /boot/efi/EFI/centos/grub.cfg
+```
+
+Else (not using UEFI):
+
+```bash
+grub2-mkconfig -o /boot/grub2/grub.cfg
+```
+
+ > Expected result
+ >
+ > Generating grub configuration file ...
+ >
+ > Found linux image: /boot/vmlinuz-3.10.0-957.el7.x86_64
+ >
+ > Found initrd image: /boot/initramfs-3.10.0-957.el7.x86_64.img
+ >
+ > Found linux image: /boot/vmlinuz-0-rescue-dcd5b66ae0534f86a683928888d3f7aa
+ >
+ > Found initrd image: /boot/initramfs-0-rescue-dcd5b66ae0534f86a683928888d3f7aa.img
+ >
+ > done
+
+
+
+#### 2. Test that the system loads with IOMMU:
+
+```bash
+dmesg | head 
```

Comment:
I wonder if we can come up with a better `grep` expression to search for, or maybe even better if there is soemthing in /proc or /sys or cpuinfo etc... we should expand/elaborate this bit.